drivers/mfd/mfd_adp5585.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2024, NXP
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT adi_adp5585
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/drivers/mfd/adp5585.h>
+
+LOG_MODULE_REGISTER(adp5585, CONFIG_GPIO_LOG_LEVEL);
+
+static int mfd_adp5585_software_reset(const struct device *dev)
+{
+	const struct mfd_adp5585_config *config = dev->config;
+	int ret = 0;
+
+	/** Set CONFIG to gpio by default */
```

Comment:
You could write both of these registers in a single transaction.

Reply:
but "i2c_burst_xx" apis are unstable, as you stated.

Reply:
You can use i2c_write like this:

```suggestion
	/** Set CONFIG to gpio by default */
	uint8_t buf[] = { ADP5585_PIN_CONFIG_A, 0, 0 };
	i2c_write_dt(&config->i2c_bus, buf, sizeof(buf));
```